src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2659,13 +2659,21 @@ private BoundExpression BindCatchFilter(CatchFilterClauseSyntax filter, Diagnost
             BoundExpression boundFilter = this.BindBooleanExpression(filter.FilterExpression, diagnostics);
             if (boundFilter.ConstantValue != ConstantValue.NotAvailable)
             {
-                Error(diagnostics, ErrorCode.WRN_FilterIsConstant, filter.FilterExpression);
+                // Depending on whether the filter constant is true or false, and whether there are other catch clauses,
+                // we suggest different actions
+                var errorCode = boundFilter.ConstantValue.BooleanValue
+                    ? ErrorCode.WRN_FilterIsConstantTrue
+                    : (filter.Parent.Parent as TryStatementSyntax).Catches.Count == 1
+                        ? ErrorCode.WRN_FilterIsConstantFalseRedundantTryCatch
```

Comment:
It would not be redundant if there is also a `finally` clause.

Reply:
That is true. It should be FilterIsConstantFalse in that case. I will make the fix.